$NetBSD: patch-configure.ac,v 1.1 2023/07/14 10:43:34 micha Exp $

Upstream expects a specific compiler/linker for a specific OS.
pkgsrc is not limited to such configurations.
Use COMPILER_RPATH_FLAG to set runtime library search path.
IRIX: OpenGL is libGL; -lgl is IRIS GL, which the OpenGL modes do not use.

--- configure.ac.orig
+++ configure.ac
@@ -3410,7 +3410,7 @@
       fi
     ;;
     *-*-irix5* | *-*-irix6* )
-      XLOCKLIBS="${XLOCKLIBS} -lgl"
+      # OpenGL is libGL; -lgl would be IRIS GL
     ;;
   esac
   ac_cv_path_opengl="no_opengl= ac_opengl_includes=$opengl_includes ac_opengl_libraries=$opengl_libraries"
@@ -5040,18 +5040,18 @@
         dirname=`which gcc`
         dirname=`dirname $dirname`
         dirname=`dirname $dirname`/lib
-	if test "$dirname" != "/lib"; then
-  	    XLOCK_LDFLAGS="${XLOCK_LDFLAGS}:$dirname"
-  	    XMLOCK_LDFLAGS="${XMLOCK_LDFLAGS}:$dirname"
-	fi
+#	if test "$dirname" != "/lib"; then
+#  	    XLOCK_LDFLAGS="${XLOCK_LDFLAGS}:$dirname"
+#  	    XMLOCK_LDFLAGS="${XMLOCK_LDFLAGS}:$dirname"
+#	fi
     fi
     case "${canonical}" in
       *-*-freebsd* )
         FORTUNE_PATH="/usr/bin/fortune"
       ;;
     esac
-    XLOCKLDFLAGS="-Wl,-R${XLOCK_LDFLAGS}"
-    XMLOCKLDFLAGS="-Wl,-R${XMLOCK_LDFLAGS}"
+    XLOCKLDFLAGS="${COMPILER_RPATH_FLAG}${XLOCK_LDFLAGS}"
+    XMLOCKLDFLAGS="${COMPILER_RPATH_FLAG}${XMLOCK_LDFLAGS}"
   ;;
 
   *-*-dgux5* )
@@ -5171,13 +5171,13 @@
         dirname=`which gcc`
         dirname=`dirname $dirname`
         dirname=`dirname $dirname`/lib
-	if test "$dirname" != "/lib"; then
-  	    XLOCK_LDFLAGS="${XLOCK_LDFLAGS}:$dirname"
-  	    XMLOCK_LDFLAGS="${XMLOCK_LDFLAGS}:$dirname"
-	fi
+#	if test "$dirname" != "/lib"; then
+#  	    XLOCK_LDFLAGS="${XLOCK_LDFLAGS}:$dirname"
+#  	    XMLOCK_LDFLAGS="${XMLOCK_LDFLAGS}:$dirname"
+#	fi
     fi
-    XLOCKLDFLAGS="-R${XLOCK_LDFLAGS}"
-    XMLOCKLDFLAGS="-R${XMLOCK_LDFLAGS}"
+    XLOCKLDFLAGS="${COMPILER_RPATH_FLAG}${XLOCK_LDFLAGS}"
+    XMLOCKLDFLAGS="${COMPILER_RPATH_FLAG}${XMLOCK_LDFLAGS}"
   ;;
 
   *-*-sunos4* )
